es/es-server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -774,10 +774,11 @@ public ClusterState execute(ClusterState currentState) {
                         ImmutableOpenMap.Builder<ShardId, ShardSnapshotStatus> shards = ImmutableOpenMap.builder();
                         boolean snapshotChanged = false;
                         for (ObjectObjectCursor<ShardId, ShardSnapshotStatus> shardEntry : snapshot.shards()) {
-                            ShardSnapshotStatus shardStatus = shardEntry.value;
+                            final ShardSnapshotStatus shardStatus = shardEntry.value;
+                            final ShardId shardId = shardEntry.key;
                             if (!shardStatus.state().completed() && shardStatus.nodeId() != null) {
```

Comment:
Aren't we missing this part here ?
https://github.com/elastic/elasticsearch/pull/47552/files#diff-a0853be4492c052f24917b5c1464003dR802-R803

Reply:
You are right, thank you.